app/services/workflow_delete_service.rb

```diff
@@ -0,0 +1,17 @@
+class WorkflowDeleteService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def destroy
```

Comment:
@bzwei @hsong-rh I feel like we need to re-sync on how we are defining service classes in this app.  At the start of the Catalog/Approval work we had talked about keeping services uniform following a structure which you can see here: https://github.com/RedHatInsights/catalog-api/blob/master/app/services/api/v1x0/catalog/add_to_order.rb

Basically, a `initialize` method if needed and a `process` method that always returned `self`.  Plus, I don't believe the service classes need to have "Service" in the name as the class names would normal define the action they were performing.  (Like the `AddToOrder` class above.) It looks like we have drifted away from this concept but I feel like it is something we should try to get back into alignment and uniform across the projects.  I'm sorry that I failed to catch this in earlier PRs.

@eclarizio Curious if you have thoughts on this topic.

Reply:
@gmcculloug We did not make all service class to implement common `process` method from day one, so it is not really a drifting away. Catalog/approval are technically two independent projects, we are thinking we can take some different approaches.

Our current service classes have their own naming conventions. All classes end with `Service` to indicate they are service classes, just like controller classes end with `Controller`. Each class has verbs to indicate what function the method is about to perform. We think `process` is too generic. More importantly we allow one service to perform multiple closely related operations. BTW, if each service class only have one `process`, what is the use of the returned `self`?

Above is our reasoning. We are willing to move to `process` style if you all think it is more appropriate.

Reply:
@gmcculloug @bzwei I don't particularly like the whole `process` that always returns `self`, but it *is* a pattern that we have followed for a while so that's kind of the only reason I can see to continue it. I think that if we ever go to some sort of service delegator or something that calls the various services it makes it easy to handle since all the services are homogenous with their main method.

In the past, I've generally done as Bill mentioned, suffixed the class with a `Service`. As I think about it more, though, it does feel like adding the `Service` suffix is a bit redundant, given that the classes themselves are in the `app/services` folder. I can go either way on adding `Service` as a suffix, since it does basically follow rail convention with controllers, and helpers, though models obviously don't have `Model` as a suffix.

I think the argument that `process` is too generic is sound, but on the other hand I think if you're limiting your service to a single responsibility (which we should be, to keep them simple), the name of the service itself should be indicative enough of what it does, and `process` is fine. You wouldn't expect the `WorkflowDeleteService` to go and do anything other than delete workflows, so `WorkflowDeleteService.new.delete` is also redundant.

> "More importantly we allow one service to perform multiple closely related operations"

I think as silly as it might seem, separating these things out into two separate services would be the way to go. Easier to test since the service is only doing one thing, and more decoupled.

Reply:
Yes in many cases one service is doing one thing, but why do we have to limit to one method? For example `create` and `create!` are very closely related, it makes sense to group them together.
For a real example, our `JbpmProcessService` has `#start` and `#signal` methods. They are very similar and even share some code.

About appending service to the class name - yes, all classes are placed under service folder which makes the appendix seem redundant. But in the code where the service class is called we can't easily tell the class is from the service folder. For readability purpose I think the appendix helps.

Reply:
@bzwei I mostly agree with what you're saying, which is why I wasn't really a huge fan of the "process" call in the first place, was just laying out some of the reasons I think that we were using that method. I think your `JbpmProcessService` may be a poor example, because sure, the methods *look* the same, but `#start` has *way* more complexity, due to using the private method `#process_options` which in turn uses `#enhance_groups`, etc. However I understand your point. I still think it would be cleaner if that logic was separated out, it feels like one class is doing too much at that point.

The only place I'd expect service classes themselves to be called is either in the controller, or, if they are more of an infrastructure service, in a model, and I don't know if it's that important that we know by the name that it is a service. I would just assume anything being called by a controller or a model is some kind of service, regardless of if it has the "Service" suffix. Again, I don't really mind that we tack on `Service` to the end, just sharing why I think we currently don't.